common/models/role.js

```diff
@@ -66,8 +77,14 @@ module.exports = function(Role) {
           roles: ACL.ROLE,
         };
 
-        var model = relsToModels[rel];
-        listByPrincipalType(this, model, relsToTypes[rel], query, callback);
+        // resolve user model and user type if principalType is available and not "USER"
+        if (rel === 'users' &&
+          query.where.principalType && query.where.principalType !== RoleMapping.USER) {
+          relsToModels.users = this.constructor.registry.getModel(query.where.principalType);
+          relsToTypes.users = query.where.principalType;
```

Comment:
I find this a bit convoluted. How about extracting the value of `relsToModels[rel]` and `relsToTypes[rel]` to two variables, and then modify only these two variables inside this branch?

```js
var targetModel = relsToModels[rel];
var targetType = relsToTypes[rel];

if (/*...*/) {
  // change property values based on query.where.principalType
}

listByPrincipalType(this, targetModel, targetType, query, callback);
```

The variable names `target*` in my snippet above are probably not very good, feel free to come with better ones.